scripts/emulate_network.go

```diff
@@ -0,0 +1,233 @@
+
+	// Save the modified content back to the file
+	//err = ioutil.WriteFile("network_params.json", modifiedContent, os.ModePerm)
+	if err != nil {
+		fmt.Println("Error writing the modified content to the file:", err)
+	}
+}
+
+func run(imageTag, imageArgs, enclaveName string, maxSteps int, kurtosisPath, kurtosisNetworkScriptFolder string, kurtosisNetConfigPath string) {
+	/*	params := map[string]interface{}{
+		"image_tag":               imageTag,
+		"image_args":              imageArgs,
+		"enclave_name":            enclaveName,
+		"max_steps":               maxSteps,
+		"kurtosis_path":           kurtosisPath,
+		"kurtosis_network_config": kurtosisNetworkScriptFolder,
+	}*/
+	update_config(imageTag, imageArgs, kurtosisNetConfigPath)
+
+	cmd := fmt.Sprintf("%s run --enclave %s %s", kurtosisPath, enclaveName, kurtosisNetworkScriptFolder)
+	runCommand(cmd)
+}
+
+func stop(kurtosisPath, enclaveName string) {
+	if enclaveName == "" {
+		fmt.Println("Error: enclave name must be specified.")
+		cmd := fmt.Sprintf("%s enclave ls", kurtosisPath)
+		runCommand(cmd)
+		return
+	}
+	cmd := fmt.Sprintf("%s enclave rm -f %s", kurtosisPath, enclaveName)
+	runCommand(cmd)
+}
+
+func runCommand(cmd string) {
+	var command *exec.Cmd
+	if runtime.GOOS == "windows" {
+		fmt.Println("Running on Windows:", cmd)
+		command = exec.Command("cmd", "/C", cmd)
+	} else {
+		fmt.Println("Running:", cmd)
+		command = exec.Command("sh", "-c", cmd)
+	}
+
+	stdoutPipe, err := command.StdoutPipe()
+	if err != nil {
+		fmt.Printf("Error obtaining stdout: %v\n", err)
+		return
+	}
+
+	stderrPipe, err := command.StderrPipe()
+	if err != nil {
+		fmt.Printf("Error obtaining stderr: %v\n", err)
+		return
+	}
+
+	err = command.Start()
+	if err != nil {
+		fmt.Printf("Command start failed: %v\n", err)
+		return
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() {
+		defer wg.Done()
+		scanner := bufio.NewScanner(stdoutPipe)
+		for scanner.Scan() {
+			fmt.Println(scanner.Text())
+		}
+	}()
+
+	go func() {
+		defer wg.Done()
+		scanner := bufio.NewScanner(stderrPipe)
+		for scanner.Scan() {
+			fmt.Println(scanner.Text())
+		}
+	}()
+
+	wg.Wait()
+
+	err = command.Wait()
+	if err != nil {
+		fmt.Printf("Command execution failed: %v\n", err)
+	}
+}
+
+func help() {
+	fmt.Println(`Emulate Network script
+Available commands:
+- build
+  - -t : Image tag (optional, default: "flashbots/builder:dev")
+  - -d : Image Build directory (optional, default: "..")
+  - -f : Build dockerfile path (optional, default: "../Dockerfile")
+- run
+  - -t : Image tag (optional, default: "flashbots/builder:dev")
+  - -n : Enclave name (optional, default: "explorer")
+  - -a : Additional builder arguments (optional)
+  - -s : Max steps (optional, default: -1)
+  - -k : Kurtosis path (optional, default: "kurtosis")
+  - -c : Kurtosis network config (optional, default: "./kurtosis")
+- stop
+  - -k : Kurtosis path (optional, default: "kurtosis")
+  - -n : Enclave name (required)
+`)
```

Comment:
"--all" Nope as it would take up on us duties of kurtosis as an enclave management tool

Regarding a single simulation "stop" acts as in docker - it will destroy a given enclave (as in stop the container)
For stopping/pausing epochs will need to develop a special command in the future